modtools/transform-unit.lua

```diff
@@ -108,7 +108,12 @@ else
  end
 
  if not caste then
-  error 'Invalid caste.'
+  local rand_caste = math.random( 0, math.abs(#race.caste-1) )
```

Comment:
Could you write a unit test to show coverage of the case where race.caste has no elements?  See the `test/` dir for sample unit tests.